### PR TITLE
[tagger] add experimental full cardinality mode

### PR DESCRIPTION
**Pull request by xvello**
### What does this PR do?

Add a `full_cardinality_tagging` undocumented option that bypasses cardinality filtering in the tagger.

**Note: this feature is experimental and not to be used by end-users as of 6.1**

Change the cardinality on the user side to keep two legit low-card uses:
  - [docker.containers.* metrics, with client-side aggregation](https://github.com/DataDog/datadog-agent/blob/81a2af3854e8f92d09b9633b83be5a6af70525e4/pkg/collector/corechecks/containers/docker.go#L91)
  - [kubernetes.pods.* metrics, ditto](https://github.com/DataDog/integrations-core/blob/eb6cb1430a29c16d52489462c9dd3c6fdd95c34b/kubelet/datadog_checks/kubelet/kubelet.py#L219)

### CODE REVIEW

File: pkg/tagger/tagger.go [hunk 1]
```diff
@@ -219,6 +222,12 @@ func (t *Tagger) Tag(entity string, highCard bool) ([]string, error) {
 	if entity == "" {
 		return nil, errors.New("empty entity ID")
 	}
+
+	// Temporary full cardinality mode override
+	if highCard == false && t.fullCard == true {
```
Comment (hkaj) [inline pkg/tagger/tagger.go]: nitpick: we don't need the first condition. Or the if actually. `highCard = highCard || t.fullCard` should do the trick

**Review by hkaj — COMMENTED** [review]
one nitpick, LGTM otherwise.

**Review by hkaj — APPROVED** [review]
Great update 💯 LGTM. @conorbranagan does it fit with your team's plan?

**Comment by conorbranagan** [comment]
yes this looks good from our side, thanks.

**Pull request merged**